[WIP] Windows Support

mihaisebea opened the pull request:
creating PR to check if it breaks linux compilation in any way

mihaisebea-gl committed:
! added support for windows
! check the readme for details

mihaisebea-gl committed:
! more examples compiling on windows
! split mesh by \n\r so it works on windows

mihaisebea-gl committed:
! moving imgui for consistency and to build it in the solution

mihaisebea-gl committed:
! see if this shows the build errors in travis

mihaisebea-gl committed:
! added imgui as a library to be consistent and to ease porting on linux
! todo check if how reloading is working on linux
! imgui example now working

mihaisebea-gl committed:
! updated path to the latest submodule folder

tgfrerer commented on CMakeLists.txt.island_prolog.in:
```diff
@@ -34,43 +34,62 @@ execute_process(COMMAND
-            COMMAND "${CMAKE_COMMAND}" --build . --target install
-            WORKING_DIRECTORY "${ISLAND_BASE_DIR}/3rdparty/src/imgui/build"
-            )
+function (link_resources)
```
this is awesome! if we had a similar method to add a link to the global resources folder that would be great - this will be useful for apps which use imgui i can imagine, and if you want to use some default shaders...

mihaisebea replied:
made the function accept parameters and now this function is reused for the global resources folder as well

tgfrerer commented on CMakeLists.txt.island_prolog.in:
```diff
@@ -34,43 +34,62 @@ execute_process(COMMAND
     # include 3rdparty includes directory
     include_directories(${ISLAND_BASE_DIR}/3rdparty/include)
 
+    IF (WIN32)
```
i think there is a way to specify an environment variable for a vulkan install - ideally, we're not hard-coding the vulkan sdk install path (i think you mention this in the readme too) perhaps if we specify that the vulkan environment variable needs to be called `VK_SDK_PATH` we could specify this here, instead.

tgfrerer commented on README.md:
```diff
@@ -279,3 +279,31 @@ reloading, these were a big initial inspiration for this project.
+
+
+Install Lunar SDK from https://vulkan.lunarg.com/
+*imgui* is disabled for now cmake was looping because of this
```
looks like this is fixed ;)

mihaisebea replied:
updated readme

tgfrerer commented on modules/le_renderer/le_renderer.cpp:
```diff
@@ -669,6 +669,7 @@ static le_resource_info_t get_default_resource_info_for_image() {
 		img.sample_count_log2 = 0; // 0 means 1, as (1 << 0 == 1)
 		img.imageType         = le::ImageType::e2D;
 		img.tiling            = le::ImageTiling::eOptimal;
+		img.samplesFlags	  = 0;
```
great catch :)

tgfrerer reviewed (approved):
Looking forward to merging this :)

mihaisebea-gl committed:
! added params to link resources function so we can link resources folder for imgui demo
! moved vs working folder outside this function since it not related
! using env variable for vulkan removing hardcoded path
! removed obsolete calls to glm
! fixed issue with window bar not visible on windows because of not taking into account the bar size
! updated README

mihaisebea-gl committed:
! updated readme with latest developments

tgfrerer commented:
awesome, thanks!
